src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,628 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import "../../../components/ha-card.js";
+import "jquery";
+import "../../../resources/roundslider.js";
+
+import { HomeAssistant } from "../../../types.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const thermostatConfig = {
+  radius: 150,
+  min: 50,
+  max: 90,
+  step: 1,
+  circleShape: "pie",
+  sliderType: "min-range",
+  value: 70,
+  startAngle: 315,
+  width: 5,
+  lineCap: "round",
+  handleSize: "+10",
+  handleShape: "dot",
+  showTooltip: true,
+};
+
+const modeColor = {
+  auto: "green",
+  heat: "#FF8100",
+  cool: "#2b9af9",
+  off: "#8a8a8a",
+};
+
+const modeIcons = {
+  auto: "mdi:autorenew",
+  heat: "hass:fire",
+  cool: "hass:snowflake",
+  off: "mdi:fan-off",
+};
+
+const thermostatState = {
+  entity_id: "climate.thermostat",
+  state: "cool",
+  attributes: {
+    current_temperature: 73,
+    min_temp: 50,
+    max_temp: 90,
+    temperature: 70,
+    target_temp_high: 75,
+    target_temp_low: 65,
+    fan_mode: "Auto Low",
+    fan_list: ["On Low", "On High", "Auto Low", "Auto High", "Off"],
+    operation_mode: "cool",
+    operation_list: ["heat", "cool", "auto", "off"],
+    hold_mode: "home",
+    swing_mode: "Auto",
+    swing_list: ["Auto", "1", "2", "3", "Off"],
+    unit_of_measurement: "°F",
+    friendly_name: "Ecobee",
+    supported_features: 1014,
+  },
+  last_changed: "2018-07-19T10:44:46.200333+00:00",
+  last_updated: "2018-07-19T10:44:46.200333+00:00",
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+}
+
+export class HuiThermostatCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+
+  static get properties() {
```

Comment:
Just realized something, we should update `shouldUpdate(changedProperties)` and make sure that `hass.states[entity_id] !== oldHass.states[entity_id]`. That way we don't re-render if it's not needed.